core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -16,17 +16,155 @@
  * @fileoverview Directive for the sidebar of the simple editor.
  */
 
+oppia.factory('ScrollSyncService', ['$anchorScroll', '$document', '$rootScope',
+  function($anchorScroll, $document, $rootScope) {
+  scrollSyncService = {
+    _targets: [],
+    _questionList: null,
+    addTarget: function(target) {
+      this._targets.push(target);
+    },
+    _clear: function() {
+      angular.element('.highlighted').removeClass('highlighted');
+    },
+    _collapse: function(hash) {
+      var question;
+      var questionIndex;
+      var questions = this._questionList.getQuestions();
+      for (var i = 0, questionsLength = questions.length;
+            i < questionsLength; i++) {
+        if (questions[i].getHash() === hash) {
+          questionIndex = i;
+          break;
+        }
+      };
+      if (!angular.isDefined(questionIndex)) {
+        return;
+      };
+      question = questions[questionIndex];
+      $rootScope.$broadcast('SimpleEditorSidebarLinkCollapse', question);
+    },
+    // Get Question id given a child id.
+    getQuestionHash: function(childHash) {
+      var hash = childHash.split('-');
+      if (hash.length > 2) {
+        return hash.slice(0, 2).join('-');
+      } else {
+        return hash[0];
+      }
+    },
+    // Update the active nav link
+    updateActive: function(hash) {
+      if (!angular.isDefined(hash)) {
+        return;
+      } else {
+        this._clear();
+        var elm = angular.element('#si-' + hash);
+
+        if (!(hash === 'intro' || hash === 'title')) {
+          elm.addClass('highlighted');
+        };
+
+        // Uncollapse the nav link in case hidden
+        if (elm.is(':hidden')) {
```

Comment:
IE has no support for `is` -- might be an issue. Is there another way to do this? Otherwise maybe we can add the necessary polyfill in app.js.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is